File: AppFinancieraIOs-Development/BackendAPI/Models/Deduction.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BackendAPI.Models
{
    public class Deduction
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();
        
        [Required]
        public DeductionType Type { get; set; }
        
        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }
        
        [Column(TypeName = "decimal(5,2)")]
        public decimal? Percentage { get; set; }
        
        [Required]
        public DateTime Date { get; set; }
        
        [MaxLength(500)]
        public string? Description { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
    
    public enum DeductionType
    {
        ISR = 0,
        IMSS = 1,
        EmploymentSubsidy = 2,
        Other = 3
    }
}

File: AppFinancieraIOs-Development/Appfinancierafuncional/Appfinancierafuncional/Models/Deduction.swift
import Foundation

struct Deduction: Identifiable, Codable {
    var id: UUID
    var type: DeductionType
    var amount: Double
    var percentage: Double?
    var date: Date
    var description: String?
    
    init(
        id: UUID = UUID(),
        type: DeductionType,
        amount: Double,
        percentage: Double? = nil,
        date: Date = Date(),
        description: String? = nil
    ) {
        self.id = id
        self.type = type
        self.amount = amount
        self.percentage = percentage
        self.date = date
        self.description = description
    }
    
    enum DeductionType: String, CaseIterable, Codable {
        case isr = "ISR"
        case imss = "IMSS"
        case employmentSubsidy = "Subsidio al Empleo"
        case other = "Otro"
        
        var icon: String {
            switch self {
            case .isr: return "dollarsign.circle.fill"
            case .imss: return "cross.circle.fill"
            case .employmentSubsidy: return "hand.raised.fill"
            case .other: return "questionmark.circle.fill"
            }
        }
    }
}
